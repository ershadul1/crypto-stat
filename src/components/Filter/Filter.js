import PropTypes from 'prop-types';
import styles from './Filter.module.css';
import currency from './currency';

const Filter = ({ filter }) => (
  <>
    <div className={styles.selector}>
      <span className={styles.label}>Currency</span>
      <select className={styles.category} onChange={e => filter({ currency: e.target.value })}>
        {currency.map(item => <option key={item} value={item}>{item.toUpperCase()}</option>)}
      </select>
    </div>
    <div className={styles.container}>
      <div onClick={() => filter({ sortBy: 'name' })} onKeyPress={() => filter({ sortBy: 'name' })} role="button" tabIndex={0}>
        Coin
        <span>&#8593;&#8595;</span>
      </div>
      <div onClick={() => filter({ sortBy: 'current_price' })} onKeyPress={() => filter({ sortBy: 'current_price' })} role="button" tabIndex={0}>
        Price
        <span>&#8593;&#8595;</span>
      </div>
      <div onClick={() => filter({ sortBy: 'market_cap' })} onKeyPress={() => filter({ sortBy: 'market_cap' })} role="button" tabIndex={0}>
        Market Cap
        <span>&#8593;&#8595;</span>
      </div>
      <div onClick={() => filter({ sortBy: 'total_volume' })} onKeyPress={() => filter({ sortBy: 'total_volume' })} role="button" tabIndex={0}>
        Total Volume
        <span>&#8593;&#8595;</span>
      </div>
      <div onClick={() => filter({ sortBy: 'price_change_percentage_24h_in_currency' })} onKeyPress={() => filter({ sortBy: 'price_change_percentage_24h_in_currency' })} role="button" tabIndex={0}>
        24H
        <span>&#8593;&#8595;</span>
      </div>
      <div onClick={() => filter({ sortBy: 'price_change_percentage_7d_in_currency' })} onKeyPress={() => filter({ sortBy: 'price_change_percentage_7d_in_currency' })} role="button" tabIndex={0}>
        7D
        <span>&#8593;&#8595;</span>
      </div>
    </div>
  </>
);

Filter.propTypes = {
  filter: PropTypes.func.isRequired,
};

export default Filter;
