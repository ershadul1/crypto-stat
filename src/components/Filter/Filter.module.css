@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@700&display=swap');

.container {
  display: flex;
  justify-content: space-around;
  border-bottom: 1px solid #000;
  padding: 30px 0;
}

.container div {
  width: 200px;
  text-align: right;
  font-family: 'Poppins', sans-serif;
  font-weight: 700;
  font-size: 18px;
  cursor: pointer;
}

.category {
  width: 22%;
  height: 45px;
  padding: 10px 15px;
  border-radius: 4px;
  border: solid 1px #e8e8e8;
  background-color: #fff;
  font-size: 20px;
}

.label {
  font-size: 20px;
  color: #000;
  font-weight: 700;
  margin-right: 10px;
}

.selector {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin: 20px 0;
}
