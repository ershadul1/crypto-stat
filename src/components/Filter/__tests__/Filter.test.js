import React from 'react';
import renderer from 'react-test-renderer';
import Filter from '../Filter';

it('Filter snapshot test', () => {
  const tree = renderer
    .create(
      <Filter filter={Function} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
