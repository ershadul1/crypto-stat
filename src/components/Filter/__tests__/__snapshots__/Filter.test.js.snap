// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter snapshot test 1`] = `
Array [
  <div
    className="selector"
  >
    <span
      className="label"
    >
      Currency
    </span>
    <select
      className="category"
      onChange={[Function]}
    >
      <option
        value="aed"
      >
        AED
      </option>
      <option
        value="ars"
      >
        ARS
      </option>
      <option
        value="aud"
      >
        AUD
      </option>
      <option
        value="bch"
      >
        BCH
      </option>
      <option
        value="bdt"
      >
        BDT
      </option>
      <option
        value="bhd"
      >
        BHD
      </option>
      <option
        value="bmd"
      >
        BMD
      </option>
      <option
        value="bnb"
      >
        BNB
      </option>
      <option
        value="brl"
      >
        BRL
      </option>
      <option
        value="btc"
      >
        BTC
      </option>
      <option
        value="cad"
      >
        CAD
      </option>
      <option
        value="chf"
      >
        CHF
      </option>
      <option
        value="clp"
      >
        CLP
      </option>
      <option
        value="cny"
      >
        CNY
      </option>
      <option
        value="czk"
      >
        CZK
      </option>
      <option
        value="dkk"
      >
        DKK
      </option>
      <option
        value="dot"
      >
        DOT
      </option>
      <option
        value="eos"
      >
        EOS
      </option>
      <option
        value="eth"
      >
        ETH
      </option>
      <option
        value="eur"
      >
        EUR
      </option>
      <option
        value="gbp"
      >
        GBP
      </option>
      <option
        value="hkd"
      >
        HKD
      </option>
      <option
        value="huf"
      >
        HUF
      </option>
      <option
        value="idr"
      >
        IDR
      </option>
      <option
        value="ils"
      >
        ILS
      </option>
      <option
        value="inr"
      >
        INR
      </option>
      <option
        value="jpy"
      >
        JPY
      </option>
      <option
        value="krw"
      >
        KRW
      </option>
      <option
        value="kwd"
      >
        KWD
      </option>
      <option
        value="lkr"
      >
        LKR
      </option>
      <option
        value="ltc"
      >
        LTC
      </option>
      <option
        value="mmk"
      >
        MMK
      </option>
      <option
        value="mxn"
      >
        MXN
      </option>
      <option
        value="myr"
      >
        MYR
      </option>
      <option
        value="ngn"
      >
        NGN
      </option>
      <option
        value="nok"
      >
        NOK
      </option>
      <option
        value="nzd"
      >
        NZD
      </option>
      <option
        value="php"
      >
        PHP
      </option>
      <option
        value="pkr"
      >
        PKR
      </option>
      <option
        value="pln"
      >
        PLN
      </option>
      <option
        value="rub"
      >
        RUB
      </option>
      <option
        value="sar"
      >
        SAR
      </option>
      <option
        value="sek"
      >
        SEK
      </option>
      <option
        value="sgd"
      >
        SGD
      </option>
      <option
        value="thb"
      >
        THB
      </option>
      <option
        value="try"
      >
        TRY
      </option>
      <option
        value="twd"
      >
        TWD
      </option>
      <option
        value="uah"
      >
        UAH
      </option>
      <option
        value="usd"
      >
        USD
      </option>
      <option
        value="vef"
      >
        VEF
      </option>
      <option
        value="vnd"
      >
        VND
      </option>
      <option
        value="xag"
      >
        XAG
      </option>
      <option
        value="xau"
      >
        XAU
      </option>
      <option
        value="xdr"
      >
        XDR
      </option>
      <option
        value="xlm"
      >
        XLM
      </option>
      <option
        value="xrp"
      >
        XRP
      </option>
      <option
        value="yfi"
      >
        YFI
      </option>
      <option
        value="zar"
      >
        ZAR
      </option>
      <option
        value="bits"
      >
        BITS
      </option>
      <option
        value="link"
      >
        LINK
      </option>
      <option
        value="sats"
      >
        SATS
      </option>
    </select>
  </div>,
  <div
    className="container"
  >
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      Coin
      <span>
        ↑↓
      </span>
    </div>
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      Price
      <span>
        ↑↓
      </span>
    </div>
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      Market Cap
      <span>
        ↑↓
      </span>
    </div>
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      Total Volume
      <span>
        ↑↓
      </span>
    </div>
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      24H
      <span>
        ↑↓
      </span>
    </div>
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      7D
      <span>
        ↑↓
      </span>
    </div>
  </div>,
]
`;
